deployments/intersmash-deployments-provider/src/main/java/org/jboss/intersmash/deployments/WildflyDeploymentApplicationConfiguration.java

```diff
@@ -0,0 +1,208 @@
+/**
+ * Copyright (C) 2023 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jboss.intersmash.deployments;
+
+import com.google.common.base.Strings;
+
+import cz.xtf.core.config.XTFConfig;
+
+/**
+ * Defines the contract for WILDFLY application services that expose configuration which should be taken into account
+ * by the WILDFLY s2i v2 process, i.e. the WildFly Maven plugin, feature packs and BOMs settings
+ */
+public interface WildflyDeploymentApplicationConfiguration {
+
+	String WILDFLY_EE_FEATURE_PACK_LOCATION = "wildfly.ee-feature-pack.location";
+	String WILDFLY_FEATURE_PACK_LOCATION = "wildfly.feature-pack.location";
+	String WILDFLY_CLOUD_FEATURE_PACK_LOCATION = "wildfly.cloud-feature-pack.location";
+	String WILDFLY_EE_CHANNEL_GROUPID = "wildfly.ee-channel.groupId";
+	String WILDFLY_EE_CHANNEL_ARTIFACTID = "wildfly.ee-channel.artifactId";
+	String WILDFLY_EE_CHANNEL_VERSION = "wildfly.ee-channel.version";
```

Comment:
perfectly aligned with CI where, on invocation, the `TESTED_WILDFLY_EAP8_EE_CHANNEL_MANIFEST` is splitted:

```shell
# e.g. org.jboss.eap.channels:eap-8.0:1.0.1.GA-redhat-20231212
if [[ "x" != "x$TESTED_WILDFLY_EAP8_EE_CHANNEL_MANIFEST" ]]; then
  arrWfChannelManifestGav=(${TESTED_WILDFLY_EAP8_EE_CHANNEL_MANIFEST//:/ })
  if [[ "x" == "x${arrWfChannelManifestGav[0]}" ]] || [[ "x" == "x${arrWfChannelManifestGav[1]}" ]] || [[ "x" == "x${arrWfChannelManifestGav[2]}" ]]; then
    echo "[ERROR] Invalid value for parameter TESTED_WILDFLY_EAP8_EE_CHANNEL_MANIFEST=$TESTED_WILDFLY_EAP8_EE_CHANNEL_MANIFEST"
    exit 1
  fi
  if [[ ! $ADDITIONAL_MAVEN_ARGS =~ '-Dwildfly.ee-channel.groupId=' ]]; then
    ADDITIONAL_MAVEN_ARGS="$ADDITIONAL_MAVEN_ARGS -Dwildfly.ee-channel.groupId=${arrWfChannelManifestGav[0]}"
  fi
  if [[ ! $ADDITIONAL_MAVEN_ARGS =~ '-Dwildfly.ee-channel.artifactId=' ]]; then
    ADDITIONAL_MAVEN_ARGS="$ADDITIONAL_MAVEN_ARGS -Dwildfly.ee-channel.artifactId=${arrWfChannelManifestGav[1]}"
  fi
  if [[ ! $ADDITIONAL_MAVEN_ARGS =~ '-Dwildfly.ee-channel.version=' ]]; then
    ADDITIONAL_MAVEN_ARGS="$ADDITIONAL_MAVEN_ARGS -Dwildfly.ee-channel.version=${arrWfChannelManifestGav[2]}"
  fi
fi
```